### PR TITLE
Refactor DisposableEventListener so it has no dependencies

### PR DESCRIPTION
**Pull request by ThijsTyZ**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -67,6 +67,7 @@
     "@babel/preset-stage-3": "^7.0.0-beta.35",
     "@babel/preset-typescript": "^7.0.0-beta.35",
     "@babel/register": "^7.0.0-beta.35",
+    "@babel/runtime": "^7.0.0-beta.35",
```
Comment (ThaNarie) [inline package.json]: What's the reason for moving this from **normal** to **dev** dependency?

The `@babel/runtime` is needed to have runtime code working, so it should be either in the `dependencies`, OR in both `devDependencies` and `peerDependencies` to make sure it's installed in the parent project.

File: src/lib/DisposableEventListener.ts [hunk 2]
```diff
@@ -1,5 +1,9 @@
-import Disposable from 'seng-disposable';
-import { EventHandler } from 'seng-event/lib/EventDispatcher';
+export type Listener = EventListenerOrEventListenerObject;
+
+export interface EventDispatcher {
+  addEventListener(type: string, listener: Listener, useCapture?: boolean): void;
+  removeEventListener(type: string, listener: Listener, useCapture?: boolean): void;
```
Comment (ThaNarie) [inline src/lib/DisposableEventListener.ts]: These typings are wrong/old:
https://developer.mozilla.org/en-US/docs/Web/API/EventTarget/addEventListener

Comment (ThaNarie) [reply src/lib/DisposableEventListener.ts]: And of course, so is our implementation.

File: package.json (hunk 1 above)
Comment (ThijsTyZ) [reply package.json]: Ok, I'll move this back

File: src/lib/DisposableEventListener.ts (hunk 2 above)
Comment (ThijsTyZ) [reply src/lib/DisposableEventListener.ts]: good point

Comment (ThijsTyZ) [reply src/lib/DisposableEventListener.ts]: Types are updated. Please review again.

File: src/lib/DisposableEventListener.ts [hunk 3]
```diff
@@ -17,51 +18,53 @@ import { EventHandler } from 'seng-event/lib/EventDispatcher';
...
+    listener: AddEventListenerParameters[1],
+    options?: AddEventListenerParameters[2],
   ) {
-    super();
-
     this.eventDispatcher = eventDispatcher;
     this.type = type;
     this.listener = listener;
-    this.useCapture = useCapture;
+    this.options = options;
```
Comment (ThaNarie) [inline src/lib/DisposableEventListener.ts:50]: Potentially assigning a legacy `useCapture` boolean to this exposed `options` object might be confusing/wrong if using it that way.

We should either expose it correctly (convert the boolean into the `options` object, or having a separate legacy `useCapture` field), or just make those/that field(s) private.

**Pull request merged**